pkg/api/api.go

```diff
@@ -193,6 +199,14 @@ func GetRuntimeInfoFunc(logger log.Logger) RuntimeInfoFn {
 	}
 }
 
+func getTenantFromHeaders(next http.Handler) http.HandlerFunc{
+	return func(w http.ResponseWriter, r *http.Request) {
+		next.ServeHTTP(w, r.WithContext(
+			context.WithValue(r.Context(), tenantIDKey, r.Header.Get("tenantIdentifier")),
```

Comment:
Please avoid hardcoding, this is probably important header name that we also need to document somewhere

Reply:
Is it worth to add empty string if the header is not set?

Reply:
We can add a flag `default-tenant-id` (Let's make sure flag name is similar to the receive)